watertap/tools/oli_api/flash.py

```diff
@@ -75,420 +73,1163 @@
+            return json_input
+
+        valid_flashes = [
+            "wateranalysis",
+            "isothermal",
+            "isenthalpic",
+            "bubblepoint",
+            "dewpoint",
+            "vapor-amount",
+            "vapor-fraction",
+            "isochoric",
+            "setph",
+            "precipitation-point",
+            "corrosion-rates",
+        ]
+        if flash_method not in valid_flashes:
+            raise RuntimeError(
+                "Invalid flash_method: {flash_method}. Use one of {', '.join(valid_flashes)}"
+            )
+
+        clone = deepcopy(json_input)
+        for k, v in survey.items():
+            d = clone["params"]
             if flash_method == "wateranalysis":
-                for param in modified_clone["params"]["waterAnalysisInputs"]:
-                    if param["name"] in survey[i].keys():
-                        param.update({"value": survey[i][param["name"]]})
-            elif flash_method == "isothermal":
-                for param in survey[i]:
-                    if param in ["Temperature", "Pressure"]:
-                        modified_clone["params"][param.lower()]["value"] = survey[i][
-                            param
-                        ]
-                    elif param in modified_clone["params"]["inflows"]["values"]:
+                d = d["waterAnalysisInputs"]
+                for param in d:
+                    if param["name"].lower() == k.lower():
                         if self.relative_inflows:
-                            modified_clone["params"]["inflows"]["values"][
-                                param
-                            ] += survey[i][param]
+                            param["value"] += v[index]
                         else:
-                            modified_clone["params"]["inflows"]["values"][
-                                param
-                            ] = survey[i][param]
-                    else:
-                        raise ValueError(
-                            "Only composition and temperature/pressure surveys are currently supported."
+                            param["value"] = v[index]
+                        _logger.info(
+                            f"Updating {k} for sample #{index} clone: new value = {param['value']}"
                         )
             else:
-                raise ValueError(
-                    " Only 'wateranalysis' and 'isothermal' currently supported."
-                )
-            clones[i] = modified_clone
-        return clones
+                if k in d:
+                    pass
+                elif k in d["inflows"]["values"]:
+                    d = d["inflows"]["values"]
+                elif k in d["corrosionParameters"]:
+                    d = d["corrosionParameters"]
+                else:
+                    _logger.warning(f"Survey key {k} not found in JSON input.")
+                if self.relative_inflows:
+                    if "value" in d[k]:
```

Comment:
The error message that I pasted in an earlier comment is attributed to this `if` logic, since `d[k]` in the cases I was testing was a float. 
```suggestion
                    if isinstance(d[k], float):
                        d[k] += v[index]
                    elif "value" in d[k]:
```

Reply:
fixed in [0900a69](https://github.com/watertap-org/watertap/pull/1296/commits/0900a69dcfa09994e9b77982c579e68b9a1d04be)